gakido/backoff.py

```
@@ -0,0 +1,142 @@
+
+
+def _default_retryable_exceptions() -> set[type[BaseException]]:
+    """Default exception types that are considered retryable."""
+    return {ConnectionError, TimeoutError, OSError}
+
+
+def _calculate_delay(attempt: int, base: float, max_delay: float, jitter: bool) -> float:
+    """Calculate exponential backoff delay with optional jitter."""
+    delay = min(base * (2 ** attempt), max_delay)
+    if jitter:
+        delay *= (0.5 + random.random() * 0.5)  # 50-100% of delay
```

Ruff (0.14.13)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)